frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -88,72 +152,330 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
               },
               {},
               deleteOptions,
-            ).catch(() => Promise.resolve()),
+            ),
           ]
         : []),
       ...(removePlugins
         ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
         : []),
     ];
+    const operandDeletionPromises = deleteOperands
+      ? operands.map((operand: K8sResourceCommon) => {
+          const model = modelFor(referenceFor(operand));
+          return k8sKill(model, operand, {}, deleteOptions);
+        })
+      : [];
+    if (operandDeletionPromises.length > 0) {
+      setOperandsDeleteInProgress(true);
+      // eslint-disable-next-line promise/catch-or-return
+      settleAllPromises(operandDeletionPromises).then(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        ([fulfilledValues, rejectedReasons, results]) => {
+          const operandsWithErrors: OperandWithErrorProps[] = results.reduce(
+            (acc: OperandWithErrorProps[], curr, i) => {
+              return curr.status === 'rejected'
+                ? acc.concat({ operand: operands[i], errorMessage: curr.reason })
+                : acc;
+            },
+            [],
+          );
+          setOperandDeletionErrors(operandsWithErrors);
+          setOperandsDeleteInProgress(false);
+          setOperandsDeleteFinished(true);
+        },
+      );
+    }
+    setOperandsDeleteFinished(!deleteOperands);
```

Comment:
Should this be...?

```suggestion
    } else {
      setOperandsDeleteFinished(true);
    }
```

I realize we hide the checkbox if there are no operands, but I'd feel better not relying on that behavior and handling the case where the checkbox is checked, but the operand list is empty.